### ISSUE TITLE
`nan` loss in examples/qm9_nnconv.py

### ISSUE THREAD

**Issue by TWRogers** [rusty1s/pytorch_geometric#783]
## 🐛 Bug

`nan` values in loss

## To Reproduce

1. Run `python examples/qm9_nnconv.py`

```bash
Epoch: 001, LR: 0.001000, Loss: nan, Validation MAE: nan, Test MAE: nan
Epoch: 002, LR: 0.001000, Loss: nan, Validation MAE: nan, Test MAE: nan
Epoch: 003, LR: 0.001000, Loss: nan, Validation MAE: nan, Test MAE: nan
...
```

## Expected behavior

Expect loss to not be `nan`

## Environment

* OS: Ubuntu `18.04`
* Python version: `3.7.5`
* PyTorch version: `1.2.0`
* CUDA/cuDNN version: `Cuda compilation tools, release 10.0, V10.0.130`
* GCC version: `gcc (Ubuntu 7.4.0-1ubuntu1~18.04.1) 7.4.0`
* Any other relevant information:

## Additional context

I am new to PyTorch and PyTorch Geometric (this is the first script I have run). 

The `nan` values in the loss seem to be as a result of the ln.23 of `torch_geometric.utils.softmax` returning large positive values (`scatter_max` returns large negative values) and then going through `out.exp()` returning `inf` values, which become `nans` in the `softmax`.

**Comment by TWRogers** [comment]
I have been able to successfully run other examples scripts such as `gcn.py` without issues.

**Comment by TWRogers** [comment]
Other package versions

```bash
torch                 1.2.0              
torch-cluster         1.4.5              
torch-geometric       1.3.2              
torch-scatter         1.3.2              
torch-sparse          0.4.3              
torch-spline-conv     1.1.1 
```

**Comment by rusty1s** [comment]
Hi, overall I cannot reproduce the `nan`s. However, I did some small changes in the `qm9_nn_conv.py` script. Can you check whether this fixes your issue?

**Comment by TWRogers** [comment]
Thanks, unfortunately I get the same issue. What versions are you using of the `torch-*` packages?

**Comment by rusty1s** [comment]
I am using all the latest releases.

**Comment by renleidewenming** [comment]
I have the same problem. Anyone help?
Downloading https://pytorch-geometric.com/datasets/qm9_v2.zip
Extracting /media/20sdb/networks/pytorch_geometric/data/QM9/raw/qm9_v2.zip
Processing...
Using a pre-processed version of the dataset. Please install `rdkit` to alternatively process the raw data.
Done!
Epoch: 001, LR: 0.001000, Loss: nan, Validation MAE: nan, Test MAE: nan
Epoch: 002, LR: 0.001000, Loss: nan, Validation MAE: nan, Test MAE: nan

**Comment by rusty1s** [comment]
Can you identify which op causes the `nan` values? Still cannot reproduce :(
```
Downloading https://pytorch-geometric.com/datasets/qm9_v2.zip
Extracting /home/matthias.fey/github/pytorch_geometric/data/QM9/raw/qm9_v2.zip
Processing...
Using a pre-processed version of the dataset. Please install 'rdkit' to alternatively process the raw data.
Done!
Epoch: 001, LR: 0.001000, Loss: 0.4960253, Validation MAE: 0.5181744, Test MAE: 0.5194247
Epoch: 002, LR: 0.001000, Loss: 0.2101828, Validation MAE: 0.3854226, Test MAE: 0.3813289
Epoch: 003, LR: 0.001000, Loss: 0.1407269, Validation MAE: 0.3225586, Test MAE: 0.3194786
Epoch: 004, LR: 0.001000, Loss: 0.0957675, Validation MAE: 0.3168485, Test MAE: 0.3150204
Epoch: 005, LR: 0.001000, Loss: 0.0681837, Validation MAE: 0.2387067, Test MAE: 0.2371089
Epoch: 006, LR: 0.001000, Loss: 0.0592947, Validation MAE: 0.2264727, Test MAE: 0.2272213
Epoch: 007, LR: 0.001000, Loss: 0.0485257, Validation MAE: 0.2195819, Test MAE: 0.2206793
Epoch: 008, LR: 0.001000, Loss: 0.0418421, Validation MAE: 0.2197249, Test MAE: 0.2206793
```
